src/QueryComplexity.ts

```diff
@@ -183,6 +191,29 @@ export default class QueryComplexity {
         (total: number, childNode: FieldNode | FragmentSpreadNode | InlineFragmentNode) => {
           let nodeComplexity = 0;
 
+          let includeNode = true;
+          let skipNode = false;
+
+          childNode.directives.forEach((directive: DirectiveNode) => {
```

Comment:
In case a server makes excessive use of directives it would iterate over all the directives, extract values, etc. regardless of whether it was already determined if we include or skip the field. 

For example: 
```graphql
{
  field @skip(if: true) @otherDirective @otherDirective2 @etc
}
```

In this case, we could skip the evaluation of the last 3 directives. I'd use `Array.find` instead and find the directives that result in the exclusion of the field.

Reply:
`@skip` and `@include` can only be used in queries/mutations from the client, and not in schema definitions on the server side. The chance of multiple directives on a query is really rare.  Also, as the `directives` are not sorted, the big O complexity is the same for both `forEach` and `find`.  If we use `forEach`, we just need to scan `directives` in one pass to get the job done.  

Directives from [the standard](http://spec.graphql.org/June2018/#sec--include):
> `directive @include(if: Boolean!) on FIELD | FRAGMENT_SPREAD | INLINE_FRAGMENT`
> `directive @skip(if: Boolean!) on FIELD | FRAGMENT_SPREAD | INLINE_FRAGMENT`